src/Compiler/CompilerPasses.cpp

```diff
@@ -84,8 +84,13 @@ void addONNXToMLIRPasses(mlir::PassManager &pm, bool targetCPU) {
   }
   // There are more opportunities for const propagation once all tensors have
   // inferred shapes.
-  pm.addNestedPass<func::FuncOp>(onnx_mlir::createConstPropONNXToONNXPass());
 
+  // We want to enable Constant Propagation only for Level O3 or when a user
+  // manually specifies the "enable-constant-prop" flag.
+  if ((/*enableConstantProp*/ optLevel >= 3 && enableConstantProp) ||
```

Comment:
I think you meant this:
```
if (optLevel >= 3 || enableConstantProp)
```

Reply:
Nope that is not what I meant... I need to update this but will explain if need be.

Reply:
The logic I have covers two cases 1) the case of when a user simply specifies the flag on the command line to enable constant propagation  and 2) when we a use Level 03 and the default flag which is set to false. So what you are proposing simply will not cover two cases but rather one.